tests/hello/hello_cancellation_test.py

```diff
@@ -0,0 +1,87 @@
+import asyncio
+import uuid
+
+from temporalio.api.common.v1 import WorkflowExecution
+from temporalio.api.workflowservice.v1 import (
+    DescribeWorkflowExecutionRequest,
+    DescribeWorkflowExecutionResponse,
+)
+from temporalio.client import Client, WorkflowExecutionStatus, WorkflowFailureError
+from temporalio.worker import Worker
+
+from hello.hello_cancellation import (
+    CancellationWorkflow,
+    cleanup_activity,
+    never_complete_activity,
+)
+
+
+async def test_cancel_workflow(client: Client):
+    task_queue_name = str(uuid.uuid4())
+
+    async with Worker(
+        client,
+        task_queue=task_queue_name,
+        workflows=[CancellationWorkflow],
+        activities=[cleanup_activity, never_complete_activity],
+    ):
+        workflow_id = str(uuid.uuid4())
+        handle = await client.start_workflow(
+            CancellationWorkflow.run,
+            id=workflow_id,
+            task_queue=task_queue_name,
+        )
+
+        await asyncio.wait_for(
+            wait_for_activity_to_start("never_complete_activity", client, workflow_id),
+            timeout=5,
+        )
+
+        await handle.cancel()
+
+        await asyncio.wait_for(
```

Comment:
@cretz I am also waiting until the workflow.status = canceled, let me know if this makes sense.

Reply:
No need to do that, just `await handle.result()` and check the exception like https://github.com/temporalio/sdk-python/blob/e37b50090392fc218bfe051063ab783c051fb0fb/tests/test_client.py#L178-L180

Reply:
No need to do that, just `await handle.result()` and check the exception like https://github.com/temporalio/sdk-python/blob/e37b50090392fc218bfe051063ab783c051fb0fb/tests/test_client.py#L178-L180

Reply:
done